KeyVault.Acmebot/Providers/TransIpProvider.cs

```diff
@@ -0,0 +1,333 @@
+            }
+
+            private async Task CreateNewToken()
+            {
+                byte[] nonce = new byte[16];
+                RandomNumberGenerator.Fill(nonce);
+
+                var request = new TokenRequest
+                {
+                    Login = _customerName,
+                    Nonce = Convert.ToBase64String(nonce)
+                };
+
+                (string signature, string body) = await SignRequestAsync(request);
+
+                var response = await new HttpClient().SendAsync(
+                    new HttpRequestMessage(HttpMethod.Post, new Uri(_httpClient.BaseAddress, "auth"))
+                    {
+                        Headers = { { "Signature", signature } },
+                        Content = new StringContent(body, Encoding.UTF8, "application/json")
+                    });
+
+                response.EnsureSuccessStatusCode();
+
+                var tokenResponse = await response.Content.ReadAsAsync<TokenResponse>();
+
+                _httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", tokenResponse.Token);
+
+                _token = new TransIPToken()
+                {
+                    CustomerName = _customerName,
+                    Token = tokenResponse.Token,
+                    Expires = DateTimeOffset.FromUnixTimeSeconds(tokenResponse.GetTokenExpiration())
+                };
+
+                StoreToken(_token);
+            }
+
+            private async Task<(string token, string body)> SignRequestAsync(object request)
+            {
+                string body = JsonConvert.SerializeObject(request);
+
+                using var hasher = SHA512.Create();
+                byte[] bytes = hasher.ComputeHash(Encoding.UTF8.GetBytes(body));
+
+                SignResult signature = await _cryptoClient.SignAsync(SignatureAlgorithm.RS512, bytes);
+
+                return (Convert.ToBase64String(signature.Signature), body);
+            }
+
+            private void StoreToken(TransIPToken token)
+            {
+                var fullPath = Environment.ExpandEnvironmentVariables(@"%HOME%\.acme\transip_token.json");
+                var directoryPath = Path.GetDirectoryName(fullPath);
+
+                if (!Directory.Exists(directoryPath))
+                {
+                    Directory.CreateDirectory(directoryPath);
+                }
+
+                var json = JsonConvert.SerializeObject(token, Formatting.Indented);
+
+                File.WriteAllText(fullPath, json);
+            }
+
+            private TransIPToken LoadToken()
+            {
+                var fullPath = Environment.ExpandEnvironmentVariables(@"%HOME%\.acme\transip_token.json");
+
+                if (!File.Exists(fullPath))
+                    return null;
+
+                var json = File.ReadAllText(fullPath);
+
+                return JsonConvert.DeserializeObject<TransIPToken>(json);
+            }
+        }
+
+        private class TransIPToken
```

Comment:
Keep the naming conventions the same as for other classes. Like `TransIpToken` in this case.